### PR TITLE
[tcgc] add `@isApiVersion` decorator

### LINKED ISSUE #1706: [tcgc] Add a decorator to signal non-standard service api version patterns
### Clear and concise description of the problem

For brownfield services with non-standard api version patterns we need a way to signal that the parameter or header is specifying the api version for the service in the generated code. We typically default api version parameters to the latest version in generated code, I would like a tcgc decorator that will signal when client libraries should treat a parameter or header like this.

For example, the storage blob service specifies and `x-ms-version` header on all requests for api versions. 

```
/** Specifies the version of the operation to use for this request. */
  @apiVersion
  @header("x-ms-version")
  version: string,
```

cc @iscai-msft 

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/azure/typespec-azure/blob/main/CODE_OF_CONDUCT.md)
- [x] Check that this issue is about the Azure libraries for typespec. For feature request in the typespec language or core libraries file it in the [TypeSpec repo](https://github.com/Microsoft/TypeSpec/issues/new/choose)
- [x] Read the [docs](https://azure.github.io/typespec-azure/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #1706

cc @jhendrixMSFT 

also, fix some typo for decorator docs.

### CODE REVIEW

File: packages/typespec-client-generator-core/test/decorators/is-api-version.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      `);
+      const sdkPackage = runner.context.sdkPackage;
+      // there will be no api version param on client, bc the service isn't versioned
+      const apiVersionClientParam = sdkPackage.clients[0].initialization.properties.find(
+        (x) => x.name === "version",
+      );
+      ok(!apiVersionClientParam);
+      const method = sdkPackage.clients[0].methods[0];
+      strictEqual(method.kind, "basic");
+      const apiVersionParam = method.parameters.find((x) => x.name === "version");
```
Comment (chunyu3) [inline packages/typespec-client-generator-core/test/decorators/is-api-version.test.ts:31]: Isn't the apiVersion parameter bumped to the client level?

Comment (tadelesh) [reply packages/typespec-client-generator-core/test/decorators/is-api-version.test.ts:31]: good catch. i think some logic is wrong for api version elevation. let me try to fix it tomorrow.

File: packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts [hunk 2]
```diff
@@ -511,18 +511,19 @@ export type ParamAliasDecorator = (
...
  * @clientNamespace("ContosoClient")
  * namespace Contoso;
  * ```
  * @example
  * ```typespec
- * @clientName("ContosoJava", "java")
- * @clientName("ContosoPython", "python")
- * @clientName("ContosoCSharp", "csharp")
- * @clientName("ContosoJavascript", "javascript")
+ * @clientNamespace("ContosoJava", "java")
```
Comment (lmazuel) [inline packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts:523]: Why is this PR including things about clientNamespace?

Comment (tadelesh) [reply packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts:523]: i just noticed this nit for the `@clientNamespace` doc and fixed it.

Comment (tadelesh) [reply packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts:523]: @lmazuel if no concern, could you also help to approve this pr?

**Review by lmazuel — APPROVED** [review]
Approve on the principal reading doc and content (didn't read the actual code)

File: packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts (hunk 2 above)
Comment (lmazuel) [reply packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts:523]: @tadelesh approved!

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**